Mathlib/Algebra/Lie/Derivation/Semisimple.lean

```diff
@@ -0,0 +1,100 @@
+/-
+Copyright © 2024 Frédéric Marbach. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Frédéric Marbach
+-/
+import Mathlib.Algebra.Lie.Derivation.AdjointAction
+import Mathlib.Algebra.Lie.Killing
+import Mathlib.LinearAlgebra.BilinearForm.Orthogonal
+
+/-!
+# Derivations of finite dimensional Killing Lie algebras
+
+This file establishes that all derivations of finite-dimensional Killing Lie algebras are inner.
+
+## Main statements
+
+The following statements hold for a finite-dimensional Lie algebra `L` with non-degenerate Killing
+form.
+
+- `LieDerivation.Killing.ad_injective`: the adjoint action is injective,
+- `LieDerivation.Killing.equiv_rangeAd`: `L` is isomorphic to its adjoint representation,
+- `LieDerivation.Killing.rangeAd_eq_top`: the range of the adjoint action is full,
+- `LieDerivation.Killing.exists_eq_ad`: any derivation is an inner derivation.
+-/
+
+namespace LieDerivation.Killing
+
+section
+
+variable (R L : Type*) [Field R] [LieRing L]
+  [LieAlgebra R L] [Module.Finite R L] [LieAlgebra.IsKilling R L]
+
+/-- A local notation for the set of (Lie) derivations on `L`. -/
+local notation "𝔻" => (LieDerivation R L L)
+
+/-- A local notation for the ideal range of `ad`. -/
+local notation "𝕀" => LieHom.idealRange (ad R L)
+
+/-- A local notation for the Killing complement of the ideal range of `ad`. -/
+local notation "𝕀ᗮ" => LieIdeal.killingCompl R 𝔻 𝕀
+
+/-- The adjoint action on a finite-dimensional Killing Lie algebra is injective. -/
+lemma ad_injective : Function.Injective (ad R L) := by
+  apply (LieHom.ker_eq_bot (ad R L)).mp
+  rw [ad_ker_eq_center, LieAlgebra.center_eq_bot_of_semisimple]
+
+/-- The Lie algebra isomorphism between a finite-dimensional Killing Lie algebra and its adjoint
+representation. -/
+noncomputable def equiv_rangeAd : L ≃ₗ⁅R⁆ (ad R L).idealRange := by
+  have h := LieEquiv.ofInjective (ad R L) (ad_injective R L)
+  rw [← ad_isIdealMorphism] at h
+  exact h
```

Comment:
I suggest:
```suggestion
lemma injective_ad_of_center_eq_bot (h : LieAlgebra.center R L = ⊥) :
    Function.Injective (ad R L) := by
  rw [← LieHom.ker_eq_bot, ad_ker_eq_center, h]

noncomputable def equivRangeAdOfCenterEqBot (h : LieAlgebra.center R L = ⊥) :
    L ≃ₗ⁅R⁆ (ad R L).idealRange :=
  letI e₁ := LieEquiv.ofInjective (ad R L) (injective_ad_of_center_eq_bot h)
  letI e₂ := LieEquiv.ofEq (ad R L).range (ad R L).idealRange (by ext; simp [mem_ad_idealRange_iff])
  e₁.trans e₂
```
The following minor points are the reasons for my suggestion:

1. These results hold for any coefficients
2. It is likely that one day we'll want to use these for Lie algebras with trivial center that do not have non-degenerate Killing form
3. We should follow the snake / camel -case conventions of our [naming conventions](https://leanprover-community.github.io/contribute/naming.html#general-conventions) (even though I don't really like these)
4. You should try never to build data using tactic mode because it produces awkward terms: tactic mode is only for `Prop` (where we don't care what the term is). The `def` here is data. More importantly because you have used an opaque `have` rather than a transparent `let`, we could never prove anything about this `def` (e.g., its behaviour when applied to an element): we would know only that it exists.